dashboard/app/controllers/congrats_controller.rb

```diff
@@ -4,6 +4,17 @@ class CongratsController < ApplicationController
   def index
     view_options(full_width: true, responsive_content: true, has_i18n: true)
 
+    begin
+      course = Base64.urlsafe_decode64(params[:s])
```

Comment:
Potential problem: If I am *on* the congrats page already, and *then* switch to a different language, I get this error:

```
NoMethodError at /congrats
==========================

undefined method `tr' for nil:NilClass

> To access an interactive console with this error, point your browser to: /__better_errors


app/controllers/congrats_controller.rb, line 8
----------------------------------------------

``` ruby
    3   
    4     def index
    5       view_options(full_width: true, responsive_content: true, has_i18n: true)
    6   
    7       begin
>   8         course = Base64.urlsafe_decode64(params[:s])
    9       rescue ArgumentError, OpenSSL::Cipher::CipherError
   10         return render status: :bad_request, json: {message: 'invalid base64'}
   11       end
   12   
   13       @next_course_script_name = ScriptConstants.csf_next_course_recommendation(course)
```
```

but then if I go back, the page suddenly is in the language that I want?

Reply:
Ideas on how to fix this?

Reply:
weird! what is the url after you switch languages -- has the `s` param disappeared? 

also, you may know this already, but to get switching languages to work, I usually have to (1) have `load_locales` set in locals.yml (and restart dashboard server), and (2) use a fresh incognito window (and also not have tried switching languages on studio.code.org in that window)

Reply:
I start at http://localhost.code.org:3000/congrats/coursed-2021?enableExperiments=studioCertificate, which redirects to http://localhost-studio.code.org:3000/congrats?s=Y291cnNlZC0yMDIx. Then I click on a new language, and it takes me to http://localhost-studio.code.org:3000/congrats?lang=en-US with an error message. If I then go back to http://localhost-studio.code.org:3000/congrats?s=Y291cnNlZC0yMDIx, it's in the new language.

So, it looks like the `s` param is getting chopped off with the new language setting

Reply:
If I make it fail safely with `params[:s] && Base64.urlsafe_decode64(params[:s])`, then it just takes them to the generic congrats page without the course-specific stuff

Reply:
I just pushed the "safe" failure ... let me know if you think I should investigate a more robust fix